### PR TITLE
Imaging uploader with phantom option

### PR DESCRIPTION
**Pull request by mohadesz**
This pull request adds a phantom drop-down with Yes and No options (By default No) on the front-end. If 'Yes' is selected, the PSCID, Candid and Visit-label options will not be required.

### CODE REVIEW

File: modules/mri_upload/php/NDB_Menu_Filter_mri_upload.class.inc [hunk 1]
```diff
@@ -118,6 +120,15 @@ class NDB_Menu_Filter_Mri_Upload extends NDB_Menu_Filter_Form
             'VisitLabel:',
             $visitlabels
         );
+        $phantom_options = array(
+                            'N' => 'No',
```
Comment (driusan) [inline modules/mri_upload/php/NDB_Menu_Filter_mri_upload.class.inc]: Shouldn't this have an empty/all option for the filter?

File: modules/mri_upload/php/NDB_Menu_Filter_mri_upload.class.inc [hunk 2]
```diff
@@ -238,39 +249,42 @@ class NDB_Menu_Filter_Mri_Upload extends NDB_Menu_Filter_Form
...
+            $visit_label = trim($args['values']['VisitLabel']);
+            if ((empty($candid)) || (empty($pscid)) || (empty($visit_label))) {
+                $errors[] ="Make sure the CandID, PSCID and Visit_label are ".
+                "filled out ";
+            }
+            $patient_name = $pscid . "_" . $candid . "_" . $visit_label;
+            ///////////////////////////////////////////////////////////////////////
+            //////check to see if candid/pscid and visit-label exist///////////////
+            ///////////////////////////////////////////////////////////////////////
+            if ((!empty($candid)) && (!empty($pscid)) && (!empty($visit_label))) {
```
Comment (driusan) [inline modules/mri_upload/php/NDB_Menu_Filter_mri_upload.class.inc]: Can't this just be an else statement for the if statement above?

File: modules/mri_upload/php/NDB_Menu_Filter_mri_upload.class.inc [hunk 3]
```diff
@@ -238,39 +249,42 @@ class NDB_Menu_Filter_Mri_Upload extends NDB_Menu_Filter_Form
...
+            ////////////// make sure the advanced options are filled out///////////
+            ///////////////////////////////////////////////////////////////////////
+            $candid      = trim($args['values']['CandID']);
+            $pscid       = trim($args['values']['PSCID']);
+            $visit_label = trim($args['values']['VisitLabel']);
+            if ((empty($candid)) || (empty($pscid)) || (empty($visit_label))) {
+                $errors[] ="Make sure the CandID, PSCID and Visit_label are ".
+                "filled out ";
+            }
+            $patient_name = $pscid . "_" . $candid . "_" . $visit_label;
```
Comment (driusan) [inline modules/mri_upload/php/NDB_Menu_Filter_mri_upload.class.inc]: This variable doesn't appear to be used.. maybe run phpmd to check for other unused variables?

**Comment by mohadesz** [comment]
This pull-request is now closed and another pull-request will be created.

**Pull request closed without merging**